### ISSUE TITLE
Check failed: read_video_result Could not load /frames/path at frame 58.

### ISSUE THREAD

**Issue by ruizhaogit** [chuckcho/video-caffe#15]
Hello,

I have compiled the video-caffe on linux.

Then, i try to train a model with some videos. 

It repots load errors for both video and frames.

Does anyone know how to solve this?

Thanks!

Best
Rui

layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc8"
  bottom: "label"
  top: "loss"
}
I0602 14:36:08.306090  8482 layer_factory.hpp:77] Creating layer data
I0602 14:36:08.306109  8482 net.cpp:91] Creating Layer data
I0602 14:36:08.306115  8482 net.cpp:399] data -> data
I0602 14:36:08.306123  8482 net.cpp:399] data -> label
I0602 14:36:08.306129  8482 data_transformer.cpp:25] Loading mean file from: /home_local/C3D/new/video-caffe/examples/ntu_rgb/ntu_train_mean.binaryproto
I0602 14:36:08.307049  8482 blob.cpp:501] legacy C3D proto detected
I0602 14:36:08.307623  8482 video_data_layer.cpp:39] Opening file /home_local/C3D/new/video-caffe/examples/ntu_rgb/c3d_ntu_test_split1.txt
I0602 14:36:08.307677  8482 video_data_layer.cpp:53] Shuffling data
I0602 14:36:08.307687  8482 video_data_layer.cpp:58] A total of 77 video chunks.
E0602 14:36:08.427530  8482 io.cpp:240] Could not read frame=72 from an image file=/home_local/C3D/new/video-caffe/examples/ntu_rgb/ntu_rgb_frames/S001C001P001R001A058_rgb/image_0072.jpg
F0602 14:36:08.427738  8482 video_data_layer.cpp:77] Check failed: read_video_result Could not load /home_local/C3D/new/video-caffe/examples/ntu_rgb/ntu_rgb_frames/S001C001P001R001A058_rgb at frame 58.
*** Check failure stack trace: ***
    @     0x7f4fe1eef40d  google::LogMessage::Fail()
    @     0x7f4fe1ef12cf  google::LogMessage::SendToLog()
    @     0x7f4fe1eeefa3  google::LogMessage::Flush()
    @     0x7f4fe1ef1bee  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f4fe2b504ec  caffe::VideoDataLayer<>::DataLayerSetUp()
    @     0x7f4fe2bf1cd3  caffe::BasePrefetchingDataLayer<>::LayerSetUp()
    @     0x7f4fe2b33595  caffe::Net<>::Init()
    @     0x7f4fe2b34435  caffe::Net<>::Net()
    @     0x7f4fe2b42c5a  caffe::Solver<>::InitTestNets()
    @     0x7f4fe2b4364d  caffe::Solver<>::Init()
    @     0x7f4fe2b4393a  caffe::Solver<>::Solver()
    @     0x7f4fe2ab8373  caffe::Creator_SGDSolver<>()
    @           0x412176  caffe::SolverRegistry<>::CreateSolver()
    @           0x40b708  train()
    @           0x40915c  main
    @     0x7f4fd2dcdb05  __libc_start_main
    @           0x40992d  (unknown)

**Comment by chuckcho** [comment]
@ruizhaogit Can you check if you have image_0072 in /home_local/C3D/new/video-caffe/examples/ntu_rgb/ntu_rgb_frames/S001C001P001R001A058_rgb? It seems it's complaining files are missing.
Note it reads 16 frames starting from the frame number specified in your training/validation frame list file.

**Comment by ruizhaogit** [comment]
@chuckcho It is there. I used the extract_frames.sh to extract the frames from the videos. I try to run the  video-caffe with very few video frames. It works. But it still does not work with lots of video frames. Does it have anything to do with the GPU memory? I use 2 GB GPU memory for now.

**Comment by chuckcho** [comment]
Try to use less batch_size, but it looks to me it's not GPU memory problem.

**Comment by ruizhaogit** [comment]
@chuckcho I use batch size as 1. 

And I try to feed some video data to the CNN net.

is gets this:

Do you have any idea about this error?

I0602 15:41:48.100859  9523 net.cpp:91] Creating Layer pool5
I0602 15:41:48.100863  9523 net.cpp:425] pool5 <- conv5a
I0602 15:41:48.100869  9523 net.cpp:399] pool5 -> pool5
I0602 15:41:48.101246  9523 net.cpp:141] Setting up pool5
I0602 15:41:48.101259  9523 net.cpp:148] Top shape: 1 256 1 4 4 (4096)
I0602 15:41:48.101264  9523 net.cpp:156] Memory required for data: 188176400
I0602 15:41:48.101269  9523 layer_factory.hpp:77] Creating layer fc6
I0602 15:41:48.101282  9523 net.cpp:91] Creating Layer fc6
I0602 15:41:48.101287  9523 net.cpp:425] fc6 <- pool5
I0602 15:41:48.101294  9523 net.cpp:399] fc6 -> fc6
E0602 15:41:48.194350  9547 io.cpp:188] not enough frames; num_frames=62, start_frame=53, length=16
F0602 15:41:48.194852  9547 video_data_layer.cpp:143] Check failed: read_video_result Could not load /home_local/nturgb+d_rgb/S001C003P003R002A053_rgb.avi at frame 53.
*** Check failure stack trace: ***
    @     0x7f839db1340d  google::LogMessage::Fail()
    @     0x7f839db152cf  google::LogMessage::SendToLog()
    @     0x7f839db12fa3  google::LogMessage::Flush()
    @     0x7f839db15bee  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f839e775f0f  caffe::VideoDataLayer<>::load_batch()
    @     0x7f839e815e19  caffe::BasePrefetchingDataLayer<>::InternalThreadEntry()
    @     0x7f839e734baf  caffe::InternalThread::entry()
    @     0x7f839e17d0cf  thread_proxy
    @     0x7f839dd3f0a4  start_thread
    @     0x7f838eab4fed  __clone

**Comment by chuckcho** [comment]
```E0602 15:41:48.194350 9547 io.cpp:188] not enough frames; num_frames=62, start_frame=53, length=16```
suggests somehow it wasn't able to find some frame(s) in 53~68 (16 frame-clip). Work with extracted frames first as video reading in OpenCV tends to be a bit fragile.

**Comment by ruizhaogit** [comment]
So, the number of frames has to be excatly divided by 16 for both input videos and input frames, right?

**Comment by chuckcho** [comment]
No. Usually the last remaining frames (15 or less) can be just disregarded.

**Comment by ruizhaogit** [comment]
Hi @chuckcho,

The image is there, but the program cannot load it. Can the profram skip this image?

How to fix this problem?

And there si another problem:  CUDNN_STATUS_SUCCESS (3 vs. 0)  CUDNN_STATUS_BAD_PARAM

Do you know how to solve this?

Thank you!

Best,
Rui

I0603 10:30:36.305829  3333 net.cpp:274] Network initialization done.
I0603 10:30:36.305946  3333 solver.cpp:60] Solver scaffolding done.
I0603 10:30:36.306535  3333 caffe.cpp:219] Starting Optimization
I0603 10:30:36.306541  3333 solver.cpp:279] Solving c3d_ntu
I0603 10:30:36.306545  3333 solver.cpp:280] Learning Rate Policy: step
I0603 10:30:36.307974  3333 solver.cpp:337] Iteration 0, Testing net (#0)
E0603 10:30:36.322075  3350 io.cpp:240] Could not read frame=54 from an image file=/home_local/nturgb+d_rgb/S001C003P008R002A054_rgb/image_0054.jpg
F0603 10:30:36.322348  3350 video_data_layer.cpp:143] Check failed: read_video_result Could not load /home_local/nturgb+d_rgb/S001C003P008R002A054_rgb at frame 54.
*** Check failure stack trace: ***
    @     0x7f8cd346d40d  google::LogMessage::Fail()
    @     0x7f8cd346f2cf  google::LogMessage::SendToLog()
    @     0x7f8cd346cfa3  google::LogMessage::Flush()
    @     0x7f8cd346fbee  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f8cd40cff0f  caffe::VideoDataLayer<>::load_batch()
    @     0x7f8cd416fe19  caffe::BasePrefetchingDataLayer<>::InternalThreadEntry()
    @     0x7f8cd408ebaf  caffe::InternalThread::entry()
F0603 10:30:36.335857  3333 cudnn_ndconv_layer.cu:43] Check failed: status == CUDNN_STATUS_SUCCESS (3 vs. 0)  CUDNN_STATUS_BAD_PARAM
*** Check failure stack trace: ***
    @     0x7f8cd346d40d  google::LogMessage::Fail()
    @     0x7f8cd3ad70cf  thread_proxy
    @     0x7f8cd346f2cf  google::LogMessage::SendToLog()
    @     0x7f8cd346cfa3  google::LogMessage::Flush()
    @     0x7f8cd36990a4  start_thread
    @     0x7f8cd346fbee  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f8cd420c16a  caffe::CudnnNdConvolutionLayer<>::Forward_gpu()
    @     0x7f8cc440efed  __clone

**Comment by ruizhaogit** [comment]
@chuckcho  Sorry, the image 54 is not in the path. It is also not in the list, why the prgram want to load it?

**Comment by ruizhaogit** [comment]
@chuckcho So, the second column in the train_spilt.txt is the number of start frame, right? Does this mean it skip the frames before the start frame?

**Comment by chuckcho** [comment]
@ruizhaogit yes. it will try to read 16 frames from the `start frame#`

**Comment by MohammadMoradi** [comment]
Hi, I have the same problem but I couldn't solve it by reducing the batch_size to 1. It reproduce the same error. Do you have any suggestion?
Thanks in advance
`...
I0722 16:36:59.101045 10645 caffe.cpp:251] Starting Optimization
I0722 16:36:59.101075 10645 solver.cpp:279] Solving c3d_ucf101
I0722 16:36:59.101084 10645 solver.cpp:280] Learning Rate Policy: step
I0722 16:36:59.105064 10645 solver.cpp:337] Iteration 0, Testing net (#0)
I0722 16:36:59.225908 10645 blocking_queue.cpp:50] Data layer prefetch queue empty
E0722 16:37:04.670671 10686 io.cpp:240] Could not read frame=208 from an image file=/home/mv_user/Data/accident_dataset/1_accident/4_accident/image_0208.jpg
F0722 16:37:04.680924 10686 video_data_layer.cpp:143] Check failed: read_video_result Could not load /home/mv_user/Data/accident_dataset/1_accident/4_accident at frame 193.
*** Check failure stack trace: ***
    @     0x7f14f13e15cd  google::LogMessage::Fail()
    @     0x7f14f13e3433  google::LogMessage::SendToLog()
    @     0x7f14f13e115b  google::LogMessage::Flush()
    @     0x7f14f13e3e1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f14f1a6524a  caffe::VideoDataLayer<>::load_batch()
    @     0x7f14f1b5f5af  caffe::BasePrefetchingDataLayer<>::InternalThreadEntry()
    @     0x7f14f1a040f5  caffe::InternalThread::entry()
    @     0x7f14e80e45d5  (unknown)
    @     0x7f14e2cf56ba  start_thread
    @     0x7f14f04383dd  clone
    @              (nil)  (unknown)
Aborted (core dumped)
`
